server/src/main/java/org/elasticsearch/common/time/DateFormatter.java

```diff
@@ -133,9 +133,11 @@ static DateFormatter forPattern(String input) {
             return Joda.forPattern(input);
         }
 
+        input = input.substring(1);
```

Comment:
This is to remove the leading `"8"` from the input pattern but I always find this confusing without a comment.